### PR TITLE
Rk development

### PR DESCRIPTION
**Pull request by RossKen**
Streamlining download and launch process

### CODE REVIEW

File: front_end/server.R [hunk 1]
```diff
@@ -14,60 +14,49 @@ function(input, output) {
       JayVees <- data.table(Document = Document, vees = vees)
       outGroup <- JayVees[, .("Similarity_score" = sum(vees)), by = Document ][order(-Similarity_score)]
       table_output <- outGroup[1:30]
-      data <- merge.data.frame(table_output, d, by.x = "Document", by.y = "Document_Number")
+      data <- merge.data.frame(table_output, data, by.x = "Document", by.y = "Document_Number")
+      data["Similarity_score"] = round(data["Similarity_score"], digits = 2)
```
Comment (EdwardAndress) [inline front_end/server.R]: Should this be a `<-` for consistency?

Comment (EdwardAndress) [inline front_end/server.R]: If we're going to use it at all, it's quite useful to be consistent because then we know, without having to refer back, where we are in the code.  I.e. `<-` is always used to assign something to a variable and `=` is used to assign function parameters.

Comment (RossKen) [inline front_end/server.R]: Yes it should be a <-... still struggling to get out of the habit of using = for variable assignment. The aim was to have it as you describe.

File: PQskmeans.R [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+lsaDim<-function(dim){return(lsaAll$tk[,1:dim] %*% diag(lsaAll$sk[1:dim]) %*% t(lsaAll$dk[,1:dim]))}
+
+#We use rank-1000 space as we are forcing 1000 clusters
+space.share<-lsaAll$tk[,1:1000] %*% diag(lsaAll$sk[1:1000]) %*% t(lsaAll$dk[,1:1000])
+
+
+#A couple of utility functions, the first gets the length of a vector, the second normalizes
+#the lengths of a matrix to length 1
+normVec<-function(vec){return(sqrt(sum(vec^2)))}
```
Comment (RossKen) [inline PQskmeans.R]: I agree that the the function name is slightly misleading. Possibly rename to 'lengthVec'?

File: PQskmeans.R [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+returnNearestMatches("domestic violence")[1:5]
+returnNearestMatches("riots in prisons")[1:5]
+returnNearestMatches("2011 riots")[1:5]
+returnNearestMatches("drug use in prisons")[1:5]
+returnNearestMatches("Employment Tribunal fees and a bunch of words that shouldn't make a difference")[1:5]
+
+
+################################
+#Code Graveyard below, but kept for reference- mostly sk means stuff, but I haven't been able to #improve on original clustering.
+################################
```
Comment (RossKen) [inline PQskmeans.R]: Cool, will do separate commit for this
